tests/cxx/operator_new.cc

```diff
@@ -55,6 +61,10 @@ void ExpressionsUserTypes() {
   IndirectClass* arr = new IndirectClass[4];
   // IWYU: IndirectClass is...*indirect.h
   delete[] arr;
+
+  decltype(elem) elem2 = nullptr;
```

Comment:
I would add a comment here to explain that the `decltype` is only used to inject sugar